File: server/src/main/java/org/tbbtalent/server/service/ProfessionService.java
package org.tbbtalent.server.service;

import org.tbbtalent.server.model.Profession;
import org.tbbtalent.server.request.profession.CreateProfessionRequest;

public interface ProfessionService {

    Profession createProfession(CreateProfessionRequest request);

    void deleteProfession(Long id);
}

File: ui/candidate-portal/src/app/services/profession.service.ts
import {Injectable} from '@angular/core';
import {environment} from "../../environments/environment";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Profession} from "../model/profession";

@Injectable({
  providedIn: 'root'
})
export class ProfessionService {

  private apiUrl: string = environment.apiUrl + '/profession';

  constructor(private http: HttpClient) { }

  createProfession(request): Observable<Profession> {
    return this.http.post<Profession>(`${this.apiUrl}`, request);
  }

  deleteProfession(id: number): Observable<any> {
    return this.http.delete<any>(`${this.apiUrl}/${id}`);
  }
}
